airflow/providers/amazon/aws/example_dags/example_sagemaker.py

```diff
@@ -0,0 +1,179 @@
+                    "S3UploadMode": "EndOfJob",
+                },
+                "AppManaged": False,
+            }
+        ]
+    },
+    "ProcessingResources": {
+        "ClusterConfig": {
+            "InstanceCount": 1,
+            "InstanceType": "ml.m5.large",
+            "VolumeSizeInGB": 5,
+        }
+    },
+    "StoppingCondition": {"MaxRuntimeInSeconds": 3600},
+    "AppSpecification": {
+        "ImageUri": f"{image_uri}",
+        "ContainerEntrypoint": ["python3", "./preprocessing.py"],
+    },
+    "RoleArn": f"{role}",
+}
+
+sagemaker_training_job_config = {
+    "AlgorithmSpecification": {
+        "TrainingImage": f"{image_uri}",
+        "TrainingInputMode": "File",
+    },
+    "InputDataConfig": [
+        {
+            "ChannelName": "config",
+            "DataSource": {
+                "S3DataSource": {
+                    "S3DataType": "S3Prefix",
+                    "S3Uri": f"s3://{s3_bucket}/config/",
+                    "S3DataDistributionType": "FullyReplicated",
+                }
+            },
+            "CompressionType": "None",
+            "RecordWrapperType": "None",
+        },
+    ],
+    "OutputDataConfig": {
+        "KmsKeyId": "",
+        "S3OutputPath": f"s3://{s3_bucket}/training/",
+    },
+    "ResourceConfig": {
+        "InstanceType": "ml.m5.large",
+        "InstanceCount": 1,
+        "VolumeSizeInGB": 5,
+    },
+    "StoppingCondition": {"MaxRuntimeInSeconds": 6000},
+    "RoleArn": f"{role}",
+    "EnableNetworkIsolation": False,
+    "EnableInterContainerTrafficEncryption": False,
+    "EnableManagedSpotTraining": False,
+    "TrainingJobName": training_job_name,
+}
+
+sagemaker_create_model_config = {
+    "ModelName": model_name,
+    "Containers": [
+        {
+            "Image": f"{image_uri}",
+            "Mode": "SingleModel",
+            "ModelDataUrl": f"s3://{s3_bucket}/training/{training_job_name}/output/model.tar.gz",
+        }
+    ],
+    "ExecutionRoleArn": f"{role}",
+    "EnableNetworkIsolation": False,
+}
+
+sagemaker_inference_config = {
+    "TransformJobName": "sample_transform_job",
+    "ModelName": model_name,
+    "TransformInput": {
+        "DataSource": {
+            "S3DataSource": {
+                "S3DataType": "S3Prefix",
+                "S3Uri": f"s3://{s3_bucket}/config/config_date.yml",
+            }
+        },
+        "ContentType": "application/x-yaml",
+        "CompressionType": "None",
+        "SplitType": "None",
+    },
+    "TransformOutput": {"S3OutputPath": f"s3://{s3_bucket}/inferencing/output/"},
+    "TransformResources": {"InstanceType": "ml.m5.large", "InstanceCount": 1},
+}
+
+# [START howto_operator_sagemaker]
+with DAG(
+    "sample_sagemaker_dag",
+    schedule_interval=None,
+    start_date=datetime(2022, 2, 21),
+    catchup=False,
+) as dag:
+    sagemaker_processing_task = SageMakerProcessingOperator(
+        config=sagemaker_processing_job_config,
+        aws_conn_id="aws_default",
```

Comment:
Out of scope for this PR, but the Operators not having a default value for `aws_conn_id` is a pain, we should fix that.

Reply:
@ferruzzi  shall we open an issue for this and might be i can take this up but I believe this will require changes in most of the operator not only related to amazon one. Right?